File: src/PristaneLaverieSmart.Application/Features/Machines/Dtos/MachineStatusAuditDto.cs
using PristaneLaverieSmart.Domain.Enums;

namespace PristaneLaverieSmart.Application.Features.Machines.Dtos;

public sealed record MachineStatusAuditDto(
    Guid Id,
    Guid MachineId,
    MachineStatus OldStatus,
    MachineStatus NewStatus,
    DateTimeOffset OccurredOn
);

File: application/src/main/java/ch/pristane/laverie/smart/application/dtos/MachineStatusAuditDto.java
package ch.pristane.laverie.smart.application.dtos;

import ch.pristane.laverie.smart.domain.enums.MachineStatus;

import java.time.OffsetDateTime;
import java.util.UUID;

public record MachineStatusAuditDto(
        UUID id,
        UUID machineId,
        MachineStatus oldStatus,
        MachineStatus newStatus,
        OffsetDateTime occurredOn
) {
}